process-kafka-quickstart-quarkus/src/test/java/org.acme.travel/MessagingIntegrationTest.java

```
@@ -0,0 +1,118 @@
+
+        kafkaTester.consume(TOPIC_CONSUMER, s -> {
+            LOGGER.info("Received from kafka: {}", s);
+            try {
+                JsonNode event = objectMapper.readValue(s, JsonNode.class);
+                Traveller traveller = objectMapper.readValue(event.get("data").toString(), Traveller.class);
+                assertTrue(traveller.isProcessed());
+                assertTrue(traveller.getFirstName().matches("Name[0-9]+"));
+                assertTrue(traveller.getLastName().matches("LastName[0-9]+"));
+                assertTrue(traveller.getEmail().matches("email[0-9]+"));
+                assertTrue(traveller.getNationality().matches("Nationality[0-9]+"));
+                countDownLatch.countDown();
+            } catch (JsonProcessingException e) {
+                LOGGER.error("Error parsing {}", s, e);
+                throw new RuntimeException(e);
+            }
+        });
+
+        IntStream.range(0, count)
+                .mapToObj(i -> new Traveller("Name" + i, "LastName" + i, "email" + i, "Nationality" + i))
+                .forEach(traveller -> kafkaTester.produce(generateCloudEvent(traveller), TOPIC_PRODUCER));
+
+        countDownLatch.await(5, TimeUnit.SECONDS);
+        assertEquals(countDownLatch.getCount(), 0);
+        kafkaTester.shutdown();
+    }
+
+    @NotNull
+    public String generateCloudEvent(Traveller traveller) {
+        assertFalse(traveller.isProcessed());
+        try {
+            return JsonNodeFactory.instance
+                    .objectNode()
+                    .put("specversion", "0.3")
+                    .put("id", UUID.randomUUID().toString())
+                    .put("source", "")
+                    .put("type", "TravelersMessageDataEvent_3")
+                    .put("time", ZonedDateTime.now().toString())
+                    .set("data", JsonNodeFactory.instance.pojoNode(traveller))
+                    .toPrettyString();
+        } catch (Exception e) {
+            return null;
```

perhaps it should propagate the exception to let the test fail?

---

dibe